core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -820,8 +822,10 @@ class ReplicaManager(val config: KafkaConfig,
           brokerTopicStats.topicStats(topicPartition.topic).messagesInRate.mark(numAppendedMessages)
           brokerTopicStats.allTopicsStats.messagesInRate.mark(numAppendedMessages)
 
-          trace(s"${records.sizeInBytes} written to log $topicPartition beginning at offset " +
-            s"${info.firstOffset.getOrElse(-1)} and ending at offset ${info.lastOffset}")
+          if (isTraceEnabled)
```

Comment:
Looking up the trace level outside of the map allows us to save a isTraceEnabled lookup for every partition being produced to.